src/c++/perf_analyzer/genai-perf/genai_perf/llm_inputs/synthetic_image_generator.py

```diff
@@ -0,0 +1,98 @@
+import base64
+from enum import Enum, auto
+from io import BytesIO
+from pathlib import Path
+from typing import List, Optional, Tuple, cast
+
+import numpy as np
+from genai_perf.exceptions import GenAIPerfException
+from PIL import Image
+
+
+class ImageFormat(Enum):
+    JPEG = auto()
+    PNG = auto()
+
+
+class RandomFormatBase64Encoder:
+    def __init__(self, image_formats: List[ImageFormat] = [ImageFormat.PNG]):
+        self.image_formats = image_formats
+
+    def __call__(self, image):
+        choice = np.random.randint(len(self.image_formats))
+        image_format = self.image_formats[choice]
+        buffered = BytesIO()
+        image.save(buffered, format=image_format.name)
+        data = base64.b64encode(buffered.getvalue()).decode("utf-8")
+        prefix = f"data:image/{image_format.name.lower()};base64"
+        return f"{prefix},{data}"
+
+
+def images_from_file_generator(image_path: Path):
+    if not image_path.exists():
+        raise GenAIPerfException(f"File not found: {image_path}")
+
+    image = Image.open(image_path)
+    while True:
+        yield image
+
+
+def white_images_generator():
+    white_image = Image.new("RGB", (100, 100), color="white")
+    while True:
+        yield white_image
+
+
+def build_synthetic_image_generator(
```

Comment:
is this a temporary code block? because I think this is essentially LlmInput's responsiblity to handle the source of data (either from file or synthetic or dataset).

Reply:
Based on other comments I simplified the `SyntheticImageGenerator`, so the builder is gone.